k8s/elasticsearch/apptest/tester/tests/basic-suite.yaml

```diff
@@ -1,8 +1,89 @@
 actions:
-- name: dummy test
+- name: Wait until all expected nodes are connected to the cluster
   bashTest:
     script: |-
-      exit 0
+      function get_nodes_count() {
+        curl -s "${HEALTH_URL}" | jq '.number_of_nodes'
+      }
+
+      expected_nodes="${REPLICAS}"
+
+      timeout=300
+      timeout_time=$(( $(date +%s) + timeout ))
```

Comment:
My personal feeling is that timeouts here are not necessary and it obscures the code. I would drop them, but this is **OPTIONAL**.

Reply:
Cluster might not be formed - do you mean not having timeouts in the test and rely on the deployer's timeout or Cloud Build's timeout? Then we rely on the execution environment for a container, which should rather be environment-agnostic.